src/features/network/store.ts

```diff
@@ -19,8 +24,28 @@ interface NetworkActions {
 
 export const useNetworkStore = create<NetworksState & NetworkActions>(
   persist(
-    set => ({
+    (set, get) => ({
       ...initialState,
+      getNetworks: () => {
+        const hostname = window.location.hostname
+        const networks = Array.from(get().byId)
+          .map(([id, network]) => ({ id, ...network }))
+          .sort(({ name: a }, { name: b }) =>
+            a.toLowerCase() < b.toLowerCase() ? -1 : 1,
+          )
+          .filter(
+            ({ filter }) =>
+              !filter ||
+              hostname.includes(filter) ||
+              hostname.includes("localhost"),
+          )
```

Comment:
All the mapping, sorting, and filtering from the view has been moved here. Note that _all_ networks are available if you're just viewing from `localhost` — but see above for a trick to set your domain to something else in local development.